### PR TITLE
pyDoc2GitHub Adding documentation for CWE-197 01 as part of #531 to GitHub

### PR DESCRIPTION
**Pull request by myteron**
Signed-off-by: Helge Wehder <helge.wehder@ericsson.com>
part of #531

### CODE REVIEW

File: docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-197/01/README.md [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+print(round(1.5))   # prints 2
+print(round(1.45))  # prints 1
+print(round(1.51))  # prints 2
+print(round(-1.5))  # prints -2
+
+print(type(round(0.5)))  # prints <class 'int'>
+
+```
+
+The build in `round()` does not allow to specify the type of rounding in use [[python round( ) 2024]](https://docs.python.org/3/library/functions.html#round). In Python 3 the `round()` function uses "bankers' rounding" (rounds to the nearest even number in case of ties). This is different to Python 2 which always rounds away from zero. Rounding provided by the decimal module allows a choice between 8 rounding modes [python decimal 2024](https://docs.python.org/3/library/decimal.html#rounding-modes). Rounding in mathematics and science is not discussed here as it requires a deeper knowledge of computer floating-point arithmetic's.
```
Comment (s19110) [inline docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-197/01/README.md:28]: cosmetics: 
- Rounding provided by the `decimal` module 
- Square brackets around sources: 
-- [[python round( ) 2024](https://docs.python.org/3/library/functions.html#round)]
-- [[python decimal 2024](https://docs.python.org/3/library/decimal.html#rounding-modes)]

File: docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-197/01/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+print(int(0.5))    # prints 0
+print(int(1.5))    # prints 1
+print(round(0.5))  # prints 0
+print(round(1.5))  # prints 2
+```
+
+## Compliant Solution (float to int)
+
+Using the `Decimal` module allows more control over rounding by choosing one of the 8 rounding modes in the decimal module.
```
Comment (s19110) [inline docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-197/01/README.md:47]: Haven't caught that in the original review, but this sentence is a bit confusing with the "Decimal" and the "decimal" modules. Maybe we could rewrite it to something like that:

Using the `Decimal` class from the `decimal` module allows more control over rounding by choosing one of the 8 rounding modes.

File: docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-197/01/README.md [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+""" Compliant Code Example """
+from decimal import Decimal, ROUND_HALF_UP, ROUND_HALF_DOWN
+ 
+print(Decimal("0.5").quantize(Decimal("1"), rounding=ROUND_HALF_UP))  # prints 1
+print(Decimal("1.5").quantize(Decimal("1"), rounding=ROUND_HALF_UP))  # prints 2
+print(Decimal("0.5").quantize(Decimal("1"), rounding=ROUND_HALF_DOWN))  # prints 0
+print(Decimal("1.5").quantize(Decimal("1"), rounding=ROUND_HALF_DOWN))  # prints 1
+```
+
+The `.quantize(Decimal("1")`, determines the precision to be integer and `rounding=ROUND_HALF_UP` determines the type of rounding applied. Specifying numbers as strings avoids issues such as floating-point representations in binary.
```
Comment (s19110) [inline docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-197/01/README.md:61]: cosmetics: `integer`

File: docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-197/01/README.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+That `Decimal` can have unexpected results when operated without `Decimal.quantize()` on floating point numbers is demonstrated in `example02.py`.
+
+[*example02.py:*](example02.py)
+
+```py
+""" Code Example """
+# SPDX-FileCopyrightText: OpenSSF project contributors
+# SPDX-License-Identifier: MIT
```
Comment (s19110) [inline docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-197/01/README.md:70]: cosmetics: license comments in the Markdown code

File: docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-197/01/README.md [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+
+## Related Guidelines
+
+|||
+|:---|:---|
+|[MITRE CWE](http://cwe.mitre.org/)|Pillar [CWE-682, Incorrect Conversion between Numeric Types (mitre.org)](http://cwe.mitre.org/data/definitions/682.html)|
+|[MITRE CWE](http://cwe.mitre.org/)|Class [CWE-197, Numeric Truncation Error](https://cwe.mitre.org/data/definitions/197.html)|
+|[SEI CERT C Coding Standard](https://wiki.sei.cmu.edu/confluence/display/c/SEI+CERT+C+Coding+Standard)|[INT31-C. Ensure that integer conversions do not result in lost or misinterpreted data](https://wiki.sei.cmu.edu/confluence/display/c/INT31-C.+Ensure+that+integer+conversions+do+not+result+in+lost+or+misinterpreted+data)|
+|[SEI CERT C Coding Standard](https://wiki.sei.cmu.edu/confluence/display/c/SEI+CERT+C+Coding+Standard)|[FLP34-C. Ensure that floating-point conversions are within range of the new type](https://wiki.sei.cmu.edu/confluence/display/c/FLP34-C.+Ensure+that+floating-point+conversions+are+within+range+of+the+new+type)|
+|[ISO/IEC TR 24772:2019]|Programming languages — Guidance to avoiding vulnerabilities in programming languages, available from [https://www.iso.org/standard/71091.html](https://www.iso.org/standard/71091.html)|
```
Comment (s19110) [inline docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-197/01/README.md:97]: The Confluence page links to web archive: [ISO/IEC TR 24772:2019](https://web.archive.org/web/20190919005645/http://www.aitcnet.org/isai/)

File: docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-197/01/README.md [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+|[SEI CERT Coding Standard for Java](https://wiki.sei.cmu.edu/confluence/display/java/SEI+CERT+Oracle+Coding+Standard+for+Java)|[NUM12-J. Ensure conversions of numeric types to narrower types do not result in lost or misinterpreted data](https://wiki.sei.cmu.edu/confluence/display/java/NUM12-J.+Ensure+conversions+of+numeric+types+to+narrower+types+do+not+result+in+lost+or+misinterpreted+data)|
+
+## Biblography
+
+|||
+|:---|:---|
+|[python round( ) 2024]|python round( ), available from: [https://docs.python.org/3/library/functions.html#round](https://docs.python.org/3/library/functions.html#round), [Last  accessed June 2024] |
+|[python decimal 2024]|Python decimal module, available from: [https://docs.python.org/3/library/decimal.html#rounding-modes](https://docs.python.org/3/library/decimal.html#rounding-modes)|
```
Comment (s19110) [inline docs/Secure-Coding-Guide-for-Python/CWE-664/CWE-197/01/README.md:105]: Missing hyperlinks:
-- [[python round( ) 2024](https://docs.python.org/3/library/functions.html#round)]
-- [[python decimal 2024](https://docs.python.org/3/library/decimal.html#rounding-modes)]

**Review by s19110 — COMMENTED** [review]
I left a few comments, mostly regarding cosmetics.

**Review by BartyBoi1128 — APPROVED** [review]
Approving this change!

**Pull request merged**